pdns/syncres.cc

```diff
@@ -2416,7 +2420,32 @@ RCode::rcodes_ SyncRes::updateCacheFromRecords(unsigned int depth, LWResult& lwr
        - NS, A and AAAA (used for infra queries)
     */
     if (i->first.type != QType::NSEC3 && (i->first.type == QType::DS || i->first.type == QType::NS || i->first.type == QType::A || i->first.type == QType::AAAA || isAA || wasForwardRecurse)) {
-      t_RC->replace(d_now.tv_sec, i->first.name, QType(i->first.type), i->second.records, i->second.signatures, authorityRecs, i->first.type == QType::DS ? true : isAA, i->first.place == DNSResourceRecord::ANSWER ? ednsmask : boost::none, recordState);
+
+      bool doCache = i->first.place != DNSResourceRecord::ANSWER || !ednsmask;
+      // if ednsmask is relevant, we do not want to cache if the scope > ecslimit and TTL < limitttl
+      if (!doCache && ednsmask) {
```

Comment:
I was a bit confused by the fact that we might still cache even though the initial value of `doCache` is `false`. How about something like that instead?
```
      bool doCache = true;
      if (i->first.place ==  DNSResourceRecord::ANSWER && ednsmask) {
```